### ISSUE TITLE
Node.js callback verify rule

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#9971]
In Node.js we would like to add a rule that allows to verify that callbacks are properly in place. This would massively help to prevent using async APIs that do not add a callback.

The `fs` module is such a case where it is planned to throw in case no callback is provided. Users would get a much better experience if they could get notified with a eslint rule.

For fs that is possible by checking if the used function contains `sync` in it's name or not and if not, if the last argument is a callback or not. Other than that I would not know how to check this. There will be multiple cases that the rule could not catch but it is probably still helpful.

See also https://github.com/nodejs/node/pull/18668/.

Do you think that is feasible?

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. I'm a bit confused about what the rule would do based on your description. Can you fill out the [rule proposal template](https://raw.githubusercontent.com/eslint/eslint/47ac478f3272fca771461e1cc6d4e91b1ecaaccc/.github/ISSUE_TEMPLATE/NEW_RULE.md) to clarify?

**Comment by BridgeAR** [comment]
**Please describe what the rule should do:**

It should warn in case a async `fs` function is used without providing a callback function.

I am not sure if this is possible at all. It was suggested in Node.js though and I thought I could just open a issue for it.

The rule could definitely not find all mistakes but probably most wrong cases.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const fs = require('fs')
fs.readFile('filename')
// Instead of
fs.readFile('filename', callback)
// Or
fs.readFileSync('filename')

fs.readFile('filename', { option: 1, option: 2 })
// Instead of
fs.readFile('filename', { option: 1, option: 2 }, () => {})

// And all other fs functions.
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It would be valuable for people working with Node.js and `fs` and hopefully at some point also other APIs than the fs module to verify that they properly added a callback in case they use a async API.
